src/lava/magma/runtime/runtime_services/runtime_service.py

```diff
@@ -295,3 +261,103 @@ def run(self):
+            self.board = N2Board(1, 1, [2], [[5, 5]])
+        else:
+            raise ValueError('Unsupported Loihi version '
+                             + 'used in board selection')
+
+    def _send_pm_cmd(self, cmd: MGMT_COMMAND):
+        for stop_send_port in self.service_to_process:
+            stop_send_port.send(cmd)
+
+    def _send_pm_rn(self, run_number: int):
+        for stop_send_port in self.service_to_process:
+            stop_send_port.send(run_number)
+
+    def _get_pm_resp(self) -> ty.Iterable[MGMT_RESPONSE]:
+        rcv_msgs = []
+        for ptos_recv_port in self.process_to_service:
+            rcv_msgs.append(ptos_recv_port.recv())
+        return rcv_msgs
+
+    def run(self):
+        self.num_steps = self.runtime_to_service.recv()
+        self.service_to_runtime.send(MGMT_RESPONSE.DONE)
+
+        selector = CspSelector()
+        channel_actions = [(self.runtime_to_service, lambda: 'cmd')]
+
+        while True:
+            action = selector.select(*channel_actions)
+            if action == 'cmd':
+                command = self.runtime_to_service.recv()
+                if enum_equal(command, MGMT_COMMAND.STOP):
+                    self._send_pm_cmd(command)
+                    rsps = self._get_pm_resp()
+                    for rsp in rsps:
+                        if not enum_equal(rsp, MGMT_RESPONSE.TERMINATED):
+                            raise ValueError(f"Wrong Response Received : {rsp}")
+                    self.service_to_runtime.send(MGMT_RESPONSE.TERMINATED)
+                    self.join()
+                    return
+                elif enum_equal(command, MGMT_COMMAND.PAUSE):
+                    self._send_pm_cmd(command)
+                    rsps = self._get_pm_resp()
+                    for rsp in rsps:
+                        if not enum_equal(rsp, MGMT_RESPONSE.PAUSED):
+                            raise ValueError(f"Wrong Response Received : {rsp}")
+
+                    self.service_to_runtime.send(MGMT_RESPONSE.PAUSED)
+                    break
+                elif enum_equal(command, MGMT_COMMAND.RUN):
+                    self._send_pm_cmd(MGMT_COMMAND.RUN)
+                    rsps = self._get_pm_resp()
+                    self._send_pm_cmd(self.num_steps)
+                    rsps = rsps + self._get_pm_resp()
+                    for rsp in rsps:
+                        if not enum_equal(rsp, MGMT_RESPONSE.DONE):
+                            raise ValueError(f"Wrong Response Received : {rsp}")
+                    self.service_to_runtime.send(MGMT_RESPONSE.DONE)
+                else:
+                    self.service_to_runtime.send(MGMT_RESPONSE.ERROR)
+
+                    self._send_pm_cmd(MGMT_COMMAND.STOP)
+                    return
+
+    def get_board(self) -> NxBoard:
+        if self.board is not None:
+            return self.board
+        else:
+            AssertionError("Cannot return board, self.board is None")
```

Comment:
Not a very meaningful assertion. There's no obvious reason why you could not return board=None. 
What's the actual reason this has to fail? Perhaps because the system has not been booted or set up yet?